ProfinetTools.Gui/ViewModels/SettingsViewModel.cs

```diff
@@ -3,31 +3,53 @@
 using System.Reactive.Linq;
 using System.Reactive.Subjects;
 using System.Reactive.Threading.Tasks;
+using System.Threading;
 using System.Threading.Tasks;
 using System.Windows;
 using ProfinetTools.Interfaces.Extensions;
 using ProfinetTools.Interfaces.Models;
 using ProfinetTools.Interfaces.Services;
-
+using ReactiveUI;
+
 namespace ProfinetTools.Gui.ViewModels
 {
-	public class SettingsViewModel : ViewModelBase
-	{
+	public class SettingsViewModel : ViewModelBase 
+    {
 		private readonly IDeviceService deviceService;
 		private readonly IAdaptersService adaptersService;
 		private readonly ISettingsService settingsService;
-		private Device device1;
-
-		public ReactiveUI.ReactiveCommand SaveCommand { get; set; }
+		private Device _selectedDevice;
+
+        private bool _signalActive = false;
+
+        private string _locateButtonText;
+        public string LocateButtonText
+        {
+            get { return _locateButtonText; }
+            set { this.RaiseAndSetIfChanged(ref _locateButtonText, value); }
+        }
+
+        private bool _isDeviceSelected;
+        public bool IsDeviceSelected
+        {
+            get { return _isDeviceSelected; }
+            set { this.RaiseAndSetIfChanged(ref _isDeviceSelected, value); }
+        }
+
+
+        public ReactiveUI.ReactiveCommand SaveCommand { get; set; }
 		public ReactiveUI.ReactiveCommand ResetCommand { get; set; }
 
+        public ReactiveUI.ReactiveCommand SavePermanentCommand { get; set; }
```

Comment:
Should we really make two commands for saving settings?
An enum for "Permanent" and "Temporary" would be enough and would avoid code duplication.